sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -2207,12 +2235,122 @@ public VirtualMachineImpl withAvailabilityZone(AvailabilityZoneId zoneId) {
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withOsDiskDeleteOptions(DeleteOptions deleteOptions) {
+        if (deleteOptions == null
+            || this.innerModel().storageProfile() == null || this.innerModel().storageProfile().osDisk() == null) {
+            return null;
+        }
+        this.innerModel().storageProfile().osDisk().withDeleteOption(diskDeleteOptionsFromDeleteOptions(deleteOptions));
+        return this;
+    }
+
     @Override
     public VirtualMachineImpl withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions deleteOptions) {
         this.primaryNetworkInterfaceDeleteOptions = deleteOptions;
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withSecondaryNetworkInterfacesDeleteOptions(DeleteOptions deleteOptions) {
+        // sets delete options for existing secondary network interfaces
+        if (this.innerModel().networkProfile() != null
+            && this.innerModel().networkProfile().networkInterfaces() != null) {
+            this.innerModel().networkProfile().networkInterfaces()
+                .stream()
+                .filter(nic -> !Objects.equals(primaryNetworkInterfaceId(), nic.id()))
+                .forEach(nic -> nic.withDeleteOption(deleteOptions));
+        }
+
+        // sets delete options for newly attached secondary network interfaces
+        this.secondaryNetworkInterfaceDeleteOptions.keySet()
+            .forEach(
+                nicId -> this.secondaryNetworkInterfaceDeleteOptions.put(nicId, deleteOptions));
+
+        // sets delete options for future secondary network interfaces to attach
+        this.secondaryNetworkInterfaceDefaultDeleteOptions = deleteOptions;
+        return this;
+    }
+
+    @Override
+    public VirtualMachineImpl withNetworkInterfaceDeleteOptions(String networkInterfaceId, DeleteOptions deleteOptions) {
+        if (!CoreUtils.isNullOrEmpty(networkInterfaceId)) {
+            if (networkInterfaceId.equals(primaryNetworkInterfaceId())) {
+                return withPrimaryNetworkInterfaceDeleteOptions(deleteOptions);
+            }
+
+            // try newly attached secondary network interfaces
+            this.secondaryNetworkInterfaceDeleteOptions.put(networkInterfaceId, deleteOptions);
```

Comment:
In create flow, the key is the key to depencency, not `networkInterfaceId`
https://github.com/Azure/azure-sdk-for-java/blob/1ac7ae4bd7141681900e9bc703ba3ab17a9c543b/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java#L1402-L1406

I am not sure whether `networkInterfaceId` here is even valid. If it is, you probably should not use `secondaryNetworkInterfaceDeleteOptions`, use a new var.

And the direct call to `put` is also suspicious? Why no validation that this 'networkInterfaceId` exists in new attached ones?